alacritty/src/config/monitor.rs

```diff
@@ -15,18 +14,34 @@ const DEBOUNCE_DELAY: Duration = Duration::from_millis(10);
 #[cfg(not(any(target_os = "linux", target_os = "macos", target_os = "windows")))]
 const DEBOUNCE_DELAY: Duration = Duration::from_millis(1000);
 
-pub fn watch(mut paths: Vec<PathBuf>, event_proxy: EventProxy) {
-    // Canonicalize all paths, filtering out the ones that do not exist.
-    paths = paths
-        .drain(..)
-        .filter_map(|path| match fs::canonicalize(&path) {
-            Ok(path) => Some(path),
-            Err(err) => {
-                error!("Unable to canonicalize config path {:?}: {}", path, err);
-                None
-            },
-        })
-        .collect();
+pub fn watch(paths: Vec<PathBuf>, event_proxy: EventProxy) {
+    let paths = {
+        let mut paths_tmp = Vec::new();
+        for path in paths {
+            if path.exists() {
+                paths_tmp.push(path.clone());
+
+                // Also watch the targets of symbolic links
+                match path.symlink_metadata() {
+                    Ok(metadata) if metadata.file_type().is_symlink() => {
```

Comment:
These operations should be chained. We shouldn't have three levels of conditionals (match -> if -> match) when all of that can be done in a single map/filter/whatever chain.

Reply:
The problem with a more functional approach is that I'd need a `flat_map` along the way (because each input path can yield 0, 1 or 2 output paths), and Rust doesn't have a `yield` keyword so it seems like a pain in the behind to generate iterators ad hoc.

I could however make the code itself flatter.

Reply:
In this case itself, there's only ever one path being yielded. I'm not talking about the loop in general, just about everything past the symlink metadata check.

Reply:
Oh, ok. I just pushed code implementing the new logic, I don't think this discussion applies anymore.